core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -119,6 +129,55 @@ public function load(array $configs, ContainerBuilder $container): void
             ->registerForAutoconfiguration(ContentCompositionInterface::class)
             ->addTag('contao.page')
         ;
+
+        $container->registerAttributeForAutoconfiguration(
+            AsContentElement::class,
+            static function (ChildDefinition $definition, AsContentElement $attribute): void {
+                $definition->addTag(ContentElementReference::TAG_NAME, $attribute->attributes);
```

Comment:
I haven't seen `$attribute->attributes` here before. Isn't it usually `get_object_vars($attribute)`?

Reply:
it can be done however we want. I found Symfony to use `get_object_vars` because they define public properties for the possible tag attributes. In the case of content elements and frontend modules, we allow all unknown properties and forward them to the fragment/renderer. In annotations, this was only possible through an array of attributes as a last-resort. With PHP though we can use unlimited named arguments for these attributes 😎